tests/BenchmarkDotNet.Tests/Exporters/XmlSerializerTests.cs

```diff
@@ -0,0 +1,245 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using BenchmarkDotNet.Exporters.Xml;
+using Xunit;
+
+namespace BenchmarkDotNet.Tests.Exporters
+{
+    public class XmlSerializerTests
+    {
+        [Fact]
+        public void Serialize()
+        {
+            string expected =
+                "Start " +
+                "MockSource " +
+                    "IsValid True IsValid "+
+                    "IntNumber 42 IntNumber " +
+                    $"DoubleNumber {double.MaxValue.ToString(CultureInfo.InvariantCulture)} DoubleNumber " +
+                    $"LongNumber {long.MaxValue} LongNumber " +
+                    "Name Source Name " +
+                    "Items " +
+                        "Item Name i1 Name Item " +
+                        "Item Name i2 Name Item " +
+                    "Items " +
+                    "DoubleArray " +
+                        "Item 1 Item " +
+                        "Item 2 Item " +
+                    "DoubleArray " +
+                "MockSource " +
+                "End";
+            var source = new MockSource();
+            var writer = new MockXmlWriter();
+            IXmlSerializer serializer = new XmlSerializer(typeof(MockSource));
+
+            serializer.Serialize(writer, source);
+            string actual = writer.ToString();
+
+            Assert.Equal(expected, actual);
```

Comment:
"There must be something bonkers with the code if it seems complicated and not self-explanatory" ;) But yeah you're right, I had a bit hard time figuring out the right way to test this method and apparently it shows too. The problem is probably just in the methodology behind "how to test the serialization".

I'm not really sure if I "went too far" with the unit testing methodology by _not allowing_ the serializer's tests to depend on any other implementation e.g. `SimpleXmlWriter`. That's why I'm using the `MockXmlWriter`, but its implementation is almost the same as `SimpleXmlWriter`'s, just without all the xml..

I tried testing the `Serialize` method just by storing each element tag call and item property addition etc. to collections inside the mock writer, and then compared those collections to expected values to define if the serialization process was successful. But after trying that out it seemed like using a string comparison would be more clean to read and work with.

One way to make this test method a bit more clear would be to actually test the xml serializer with some real xml data. That would mean `XmlSerializerTests` would have to implement `IDisposable` (`SimpleXmlWriter` is disposable) and then handle disposing accordingly.

If your first reaction when seeing the `Serialize` test was "_Where's the xml?_", then it's really an easy decision to change the expected value of `Serialize` test from current custom string to xml before accepting this PR. Should I do it or leave it as is?

(I could have had use for your white space implementation here, now that I have looked into it)